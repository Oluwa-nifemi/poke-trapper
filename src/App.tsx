import React from 'react';
import './App.css';
import {Route, Routes} from "react-router-dom";
import Layout from "./components/layout";
import ListPokemon from "./pages/list-pokemon";
import PokemonDetails from "./pages/pokemon-details";
import MyPokemonList from "./pages/my-pokemon-list";

function App() {
    return (
        <Routes>
            <Route path="/" element={<Layout/>}>
                <Route index element={<ListPokemon/>}/>
                <Route path="/:id" element={<PokemonDetails/>}/>
                <Route path="/my-pokemon" element={<MyPokemonList/>}/>
            </Route>
        </Routes>
    );
}

export default App;
