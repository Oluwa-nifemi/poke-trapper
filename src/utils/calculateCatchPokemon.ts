const calculateCatchPokemon = (): boolean => Math.random() > 0.5

export default calculateCatchPokemon