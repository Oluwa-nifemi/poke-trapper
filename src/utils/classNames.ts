/* This function allows you to merge several classes together.
 * For conditional classes the first item in the array should be the condition (boolean)
 */
export const classNames = (...args: (string | [boolean, string])[]) => {
    return args.filter(className => {
        if (Array.isArray(className)) {
            return className[0];
        }

        return className;
    }).map((className) => {
        if (Array.isArray(className)) {
            return className[1];
        }

        return className;
    }).join(' ')
};