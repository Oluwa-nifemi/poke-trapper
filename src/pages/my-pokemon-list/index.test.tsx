import {render, screen} from "tests/utils";
import App from "../../App";
import localforage from "localforage";
import {waitFor} from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("My Pokemon List", () => {
    beforeEach(() => {
        window.history.pushState({}, "", "/my-pokemon");
    })

    it("Renders empty message", () => {
        render(<App/>);

        expect(screen.getByText("You have no saved pokemon yet.")).toBeInTheDocument();
    })

    it("Renders list of pokemon", async () => {
        const testPokemonNickname = "Test Pokemon";

        //Update local storage with data
        await localforage.setItem("caughtPokemon", [
            {
                id: "123",
                nickname: testPokemonNickname,
                sprites: {
                    front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                },
                name: "bulbasaur"
            }
        ])

        await localforage.setItem("stats", {
            bulbasaur: 1
        })

        const {container} = render(<App/>);

        await waitFor(() => {
            expect(screen.queryByText("Extracting local database...")).toBeNull();
        })

        expect(container).toMatchSnapshot()

        const user = userEvent.setup()

        await user.click(screen.getByText("Release"))

        expect(container).toMatchSnapshot()
        expect(screen.getByTitle(`${testPokemonNickname} has been released!`)).toBeInTheDocument()
    })
})