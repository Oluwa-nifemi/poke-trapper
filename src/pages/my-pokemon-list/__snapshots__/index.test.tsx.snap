// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Pokemon List Renders list of pokemon 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      <svg
        class="navLogo"
      >
        logo.svg
      </svg>
    </a>
    <a
      class="navLink"
      href="/"
    >
      Home
    </a>
  </nav>
  <main
    class="main"
  >
    <div>
      <h1
        class="header"
      >
        My Pokemon
      </h1>
      <ul
        class="myPokemonList"
      >
        <li
          class="myPokemonListItem"
        >
          <a
            class="myPokemonListItemLink"
            href="/bulbasaur"
          >
            <h2
              class="myPokemonListItemNickname"
            >
              Test Pokemon
            </h2>
            <h3
              class="myPokemonListItemName"
            >
              bulbasaur
            </h3>
          </a>
          <img
            alt="bulbasaur"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
          />
          <button
            class="pixelatedButton myPokemonListItemRelease"
          >
            <span
              class="pixelatedButtonText"
            >
              Release
            </span>
          </button>
        </li>
      </ul>
      <div
        class="popup"
        role="alert"
      />
    </div>
  </main>
</div>
`;

exports[`My Pokemon List Renders list of pokemon 2`] = `
<div>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      <svg
        class="navLogo"
      >
        logo.svg
      </svg>
    </a>
    <a
      class="navLink"
      href="/"
    >
      Home
    </a>
  </nav>
  <main
    class="main"
  >
    <div>
      <h1
        class="header"
      >
        My Pokemon
      </h1>
      <h2
        class="emptyMessage"
      >
        <span>
          You have no saved pokemon yet.
        </span>
        <br />
        <span>
          Go Catch Some!
        </span>
      </h2>
      <div
        class="popup isVisible"
        role="alert"
      >
        <p
          title="Test Pokemon has been released!"
        >
          <span
            class="letter"
            style="animation-delay: 0s;"
          >
            T
          </span>
          <span
            class="letter"
            style="animation-delay: 0.06s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 0.12s;"
          >
            s
          </span>
          <span
            class="letter"
            style="animation-delay: 0.18s;"
          >
            t
          </span>
          <span
            class="letter"
            style="animation-delay: 0.24s;"
          >
             
          </span>
          <span
            class="letter"
            style="animation-delay: 0.3s;"
          >
            P
          </span>
          <span
            class="letter"
            style="animation-delay: 0.36s;"
          >
            o
          </span>
          <span
            class="letter"
            style="animation-delay: 0.42s;"
          >
            k
          </span>
          <span
            class="letter"
            style="animation-delay: 0.48s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 0.54s;"
          >
            m
          </span>
          <span
            class="letter"
            style="animation-delay: 0.6s;"
          >
            o
          </span>
          <span
            class="letter"
            style="animation-delay: 0.6599999999999999s;"
          >
            n
          </span>
          <span
            class="letter"
            style="animation-delay: 0.72s;"
          >
             
          </span>
          <span
            class="letter"
            style="animation-delay: 0.78s;"
          >
            h
          </span>
          <span
            class="letter"
            style="animation-delay: 0.84s;"
          >
            a
          </span>
          <span
            class="letter"
            style="animation-delay: 0.8999999999999999s;"
          >
            s
          </span>
          <span
            class="letter"
            style="animation-delay: 0.96s;"
          >
             
          </span>
          <span
            class="letter"
            style="animation-delay: 1.02s;"
          >
            b
          </span>
          <span
            class="letter"
            style="animation-delay: 1.08s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 1.14s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 1.2s;"
          >
            n
          </span>
          <span
            class="letter"
            style="animation-delay: 1.26s;"
          >
             
          </span>
          <span
            class="letter"
            style="animation-delay: 1.3199999999999998s;"
          >
            r
          </span>
          <span
            class="letter"
            style="animation-delay: 1.38s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 1.44s;"
          >
            l
          </span>
          <span
            class="letter"
            style="animation-delay: 1.5s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 1.56s;"
          >
            a
          </span>
          <span
            class="letter"
            style="animation-delay: 1.6199999999999999s;"
          >
            s
          </span>
          <span
            class="letter"
            style="animation-delay: 1.68s;"
          >
            e
          </span>
          <span
            class="letter"
            style="animation-delay: 1.74s;"
          >
            d
          </span>
          <span
            class="letter"
            style="animation-delay: 1.7999999999999998s;"
          >
            !
          </span>
        </p>
        <button
          class="button popupButton"
        >
          Close
        </button>
      </div>
    </div>
  </main>
</div>
`;
